knimin/lib/data_access.py

```diff
@@ -3191,6 +3186,252 @@ def delete_sample_plate(self, id):
+
+        Returns
+        -------
+        list of str
+            Sorted list of emails
+        """
+        with TRN:
+            sql = """SELECT email
+                     FROM ag.labadmin_users
+                     ORDER BY email"""
+            TRN.add(sql)
+            return TRN.execute_fetchflatten()
+
+    def get_sample_plate_ids(self):
+        """Gets a list of sample plate IDs
+
+        Returns
+        -------
+        list of int
+            Sorted list of sample plate IDs
+        """
+        with TRN:
+            sql = """SELECT sample_plate_id
+                     FROM pm.sample_plate
+                     ORDER BY sample_plate_id"""
+            TRN.add(sql)
+            return TRN.execute_fetchflatten()
+
+    def get_sample_plate_list(self):
+        """Gets basic information of all sample plates
+
+        Returns
+        -------
+        list of dict
+            {id : int, name : str, type : list of [str, int], count : int,
+            person : str, date : datetime,
+            study : list of [int, int, int, str])}
+            Plate id, plate name, plate type (name and total number of wells),
+            number of samples filled, email, date, study (number of studies,
+            ID, Qiita ID and title of the most frequent study)
+        """
+        with TRN:
+            sql = """SELECT sample_plate_id, sample_plate.name, plate_type.name,
+                            cols, rows, email, created_on, x.sample_count,
+                            x.study_freq, x.study_id, x.qiita_study_id, x.title
+                     FROM pm.sample_plate
+                     JOIN pm.plate_type USING (plate_type_id)
+                     JOIN (SELECT study_id, qiita_study_id, title,
+                                  sample_plate_id,
+                                  COUNT(study_id) AS study_freq,
+                                  COUNT(sample_id) AS sample_count
+                           FROM pm.study
+                           JOIN pm.study_sample USING (study_id)
+                           JOIN pm.sample_plate_layout USING (sample_id)
+                           JOIN pm.sample_plate USING (sample_plate_id)
+                           GROUP BY study_id, sample_plate_id
+                           ORDER BY COUNT(study_id) DESC) AS x
+                     USING (sample_plate_id)
+                     ORDER BY sample_plate_id"""
+            TRN.add(sql)
+            res = TRN.execute_fetchindex()
+            plates = []
+            for row in res:
+                plates.append({'id': int(row[0]),
+                               'name': row[1],
+                               'type': [row[2], row[3]*row[4]],
+                               'person': row[5],
+                               'date': row[6],
+                               'count': row[7],
+                               'study': [row[8], row[9], row[10], row[11]]})
+            return plates
+
+    def populate_plate_mapper(self):
```

Comment:
I don't think this function is needed. I think this function needs to be part of the initial SQL, since it is populating the tables with some information, and it shouldn't be a function since this only needs to be executed once the SQL tables for the plate mapper is created.